compiler/int128.nim

```diff
@@ -450,6 +450,8 @@ proc addInt128*(result: var string; value: Int128) =
       j -= 1
 
 proc `$`*(a: Int128): string =
+  # "-170141183460469231731687303715884105728".len == 41
+  result = newStringOfCap(41)
```

Comment:
tbh, this looks pretty dubious - for the vast majority of integers it will overallocate by a significant amount - int128 is used in the compiler to represent 64-bit values at most, and most such values naturally fall in the few-digits range - without benchmarks, this looks like a premature optimization that adds memory overhead without any actual performance benefit..